flask/main.py

```diff
@@ -78,105 +78,193 @@ def similarSearch(search, category):
     
     return outputstring[:-4] + '?'
 
-def findLocationAndMeal(req_data):
-    """Dialogflow ``findLocationAndMeal`` intent handler. Checks for valid Location and Meal and sends HTTP response with appropriate data.
 
-    :param req_data: Dialogflow POST request data
+
+def addFollowupEventInput(responsedata, outputParams):
+    """Helper function for adding followupEventInput trigger to send data to queryHelper intent.
+
+    :param responsedata: The HTTP response data
+    :type responsedata: dict
+    :param outputParams: Output context paramaters to send to queryHelper
+    :type outputParams: dict
+    """
+    responsedata['followupEventInput'] = {
+        'name': 'queryHelperEvent',
+        'parameters': outputParams
+    }
+    return responsedata
+
+def requiredEntitiesHandler(req_data, intentname):
+    """Handles response for intents with 2 entities that are both required and need to be handled manually, returns approriate parameters for specific intent handler to take care of.
+
+    :param req_data: The HTTP response data
     :type req_data: JSON
+    :param intentname: Output context paramaters to send to queryHelper
+    :type intentname: string
     """
-    parameters = {}
-    
-    locationEntered = False
-    mealEntered = False
-    meal_in = ""
-    loc_in = ""    
+    if intentname == 'findLocationAndMeal':
+        entity1 = 'Location'
```

Comment:
My only thought here is that maybe this should be a list of simple entity classes. For instance

```
class Entity:
   def __init(self, name, output_context, question):
   self.name = name
   self.output_context = output_context
   self.question = question
```

Then put these all into a list?

`entities = [Entity('Location','LocationOutputContext','Which dining location'), Entity('Meal','MealOutputContext','What meal would you like?')`

Then you can just do a for loop to see if any of the entities are in the inputParams or if any are available. 

This would allow for future entities in the string. Whenever I see variables with names in them, I'm always thinking they should be a list instead. Minimally you can just change it to a list now and make a class later.

```
entities = ['Location', 'Meal']
entityOutputContext = ['LocationOutputContext', 'MealOutputContext']
```

I think this is okay as-is, maybe create this as another issue if you'd like instead, just seemed like it's clean up a little of this code.

Reply:
Gotcha, I'll close this for now and clean it up in the upcoming PR.